homeassistant/components/fan/__init__.py

```diff
@@ -238,6 +240,9 @@ def set_direction(self: ToggleEntity, direction: str) -> None:
 
     def turn_on(self: ToggleEntity, speed: str=None, **kwargs) -> None:
         """Turn on the fan."""
+        if speed is SPEED_OFF:
+            self.turn_off()
```

Comment:
Calling `turn_on` should not trigger a call to `turn_off`. I think that `SPEED_OFF` should be removed completely, that's what `turn_off` is for.